app/src/main/java/com/example/seonoh/seonohapp/MainActivity.kt

```diff
@@ -0,0 +1,123 @@
+package com.example.seonoh.seonohapp
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.util.Log
+import android.widget.Toast
+import com.example.seonoh.seonohapp.model.Market
+import com.example.seonoh.seonohapp.network.MarketRequest
+import com.google.android.material.tabs.TabLayout
+import kotlinx.android.synthetic.main.activity_main.*
+
+class MainActivity : AppCompatActivity(),MarketRequest.ResultListener {
+
+    companion object{
+        var krwMarketData : String = ""
+        var btcMarketData : String = ""
+        var ethMarketData : String = ""
+        var usdtMarketData : String = ""
+    }
+    lateinit var mPagerAdapter : TabPagerAdapter
+    private val TAG = "COIN_MAIN"
+    private val TAB_COUNT = 4
+    lateinit var  toast : Toast
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        toast  = Toast.makeText(this, "뒤로가기를 한번 더 누르시면 앱이 종료됩니다.", Toast.LENGTH_SHORT)
+
+
+        MarketRequest().send(this@MainActivity)
+
+    }
+
+    fun initView(){
+        mPagerAdapter = TabPagerAdapter(supportFragmentManager, TAB_COUNT)
+        coinViewPager.adapter = mPagerAdapter
+
+
+        coinViewPager.addOnPageChangeListener(object : TabLayout.TabLayoutOnPageChangeListener(tabLayout) {
+            override fun onPageSelected(position: Int) {
+                super.onPageSelected(position)
+
+                when (position) {
+                    0 -> {
+                        Log.e(TAG,"KRW")
+                    }
+
+                    1 -> {
+                        Log.e(TAG,"BTC")
+
+                    }
+
+                    2 -> {
+                        Log.e(TAG,"ETH")
+
+                    }
+                    3 -> {
+                        Log.e(TAG,"USDT")
+
+                    }
+
+
+                }
+
+            }
+
+
+        })
+
+        tabLayout.apply {
+            addTab(tabLayout.newTab().setText("KRW"))
+            addTab(tabLayout.newTab().setText("BTC"))
+            addTab(tabLayout.newTab().setText("ETH"))
+            addTab(tabLayout.newTab().setText("USDT"))
+        }
+
+        tabLayout.addOnTabSelectedListener(TabLayout.ViewPagerOnTabSelectedListener(coinViewPager))
```

Comment:
```suggestion
        tabLayout. setupWithViewPager(coinViewPager)
```
TabLayout - setupWithViewPager() 함수로 TabLayout - ViewPager를 연결하는 방법도 있습니다.